libethash-cl/ethash_cl_miner.cpp

```diff
@@ -78,23 +78,37 @@ ethash_cl_miner::~ethash_cl_miner()
 	finish();
 }
 
-string ethash_cl_miner::platform_info(unsigned _platformId, unsigned _deviceId)
+std::vector<cl::Platform> ethash_cl_miner::getPlatforms()
 {
 	vector<cl::Platform> platforms;
-	cl::Platform::get(&platforms);
-	if (platforms.empty())
+	try
 	{
-		ETHCL_LOG("No OpenCL platforms found.");
-		return string();
+		cl::Platform::get(&platforms);
 	}
+	catch(cl::Error const& err)
+	{
+		#if defined(CL_PLATFORM_NOT_FOUND_KHR)
```

Comment:
Maybe don't indent the preprocessor commands, since they are more readable this way? I am not sure what our style guide says about that but I personally find it more readable if they are not indented. Something like:

```
#if defined(CL_PLATFORM_NOT_FOUND_KHR)
        if (err.err() == CL_PLATFORM_NOT_FOUND_KHR)
            ETHCL_LOG("No OpenCL platforms found");
        else
#endif
            throw err;
```